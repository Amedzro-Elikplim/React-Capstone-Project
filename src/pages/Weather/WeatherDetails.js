import './weather.css';
import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useLocation, useNavigate } from 'react-router-dom';
import {
  FiCloudDrizzle, FiSun, FiCloud, FiArrowLeft,
} from 'react-icons/fi';
import { FETCHTHUNK } from '../../redux/Weather/ActionCreators';

function WeatherDetails() {
  const location = useLocation();
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const id = location.state;

  useEffect(() => {
    dispatch(FETCHTHUNK(id));
  }, []);

  function moveBack() {
    navigate(-1);
  }

  const details = useSelector((state) => state.weather);

  return (
    <>
      <div className="icon-container">
        <FiArrowLeft className="icon-left" onClick={() => moveBack()} />
      </div>
      <div className="container-fluid container-parent d-flex justify-content-center align-items-center">
        <div className="card card-container p-2 m-2">
          <div className="d-flex align-items-center gap-4">
            <div className="d-flex gap-1 align-items-center justify-content-center">
              <h1>
                {details.name}
                ,
              </h1>
            </div>
            <FiCloud />
            <FiCloudDrizzle />
            <FiSun />
          </div>

          <div className="d-flex gap-2">
            <div>
              <p>latitude</p>
              <p className="details">{details.lat}</p>
            </div>
            <div>
              <p>longitude</p>
              <p className="details">{details.log}</p>
            </div>
            <div>
              <p>temp</p>
              <p className="details">{details.temp}</p>
            </div>
            <div>
              <p>humid</p>
              <p className="details">{details.humidity}</p>
            </div>
            <div>
              <p>pressure</p>
              <p className="details">{details.pressure}</p>
            </div>
          </div>

          <div className="d-flex gap-2">
            <div>
              <p>wind speed</p>
              <p className="details">{details.wind_speed}</p>
            </div>
            <div>
              <p>sunrise</p>
              <p className="details">{details.sunrise}</p>
            </div>
            <div>
              <p>sunset</p>
              <p className="details">{details.sunset}</p>
            </div>
            <div>
              <p>feels_like</p>
              <p className="details">{details.humidity}</p>
            </div>
          </div>

          <div className="d-flex gap-3 align-items-center">
            <h3>Description</h3>
            {details.description}
          </div>
        </div>
      </div>
    </>
  );
}

export default WeatherDetails;
