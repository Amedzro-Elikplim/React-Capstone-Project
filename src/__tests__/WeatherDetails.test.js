import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { HashRouter } from 'react-router-dom';
import store from '../redux/ConfigureStore';
import WeatherDetails from '../pages/Weather/WeatherDetails';

test('it should render weather page', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <HashRouter>
          <WeatherDetails />
        </HashRouter>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
