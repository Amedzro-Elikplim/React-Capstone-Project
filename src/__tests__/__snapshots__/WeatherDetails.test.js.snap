// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render weather page 1`] = `
Array [
  <div
    className="icon-container"
  >
    <svg
      className="icon-left"
      fill="none"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="19"
        x2="5"
        y1="12"
        y2="12"
      />
      <polyline
        points="12 19 5 12 12 5"
      />
    </svg>
  </div>,
  <div
    className="container-fluid container-parent d-flex justify-content-center align-items-center"
  >
    <div
      className="card card-container p-2 m-2"
    >
      <div
        className="d-flex align-items-center gap-4"
      >
        <div
          className="d-flex gap-1 align-items-center justify-content-center"
        >
          <h1>
            ,
          </h1>
        </div>
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18 10h-1.26A8 8 0 1 0 9 20h9a5 5 0 0 0 0-10z"
          />
        </svg>
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="8"
            x2="8"
            y1="19"
            y2="21"
          />
          <line
            x1="8"
            x2="8"
            y1="13"
            y2="15"
          />
          <line
            x1="16"
            x2="16"
            y1="19"
            y2="21"
          />
          <line
            x1="16"
            x2="16"
            y1="13"
            y2="15"
          />
          <line
            x1="12"
            x2="12"
            y1="21"
            y2="23"
          />
          <line
            x1="12"
            x2="12"
            y1="15"
            y2="17"
          />
          <path
            d="M20 16.58A5 5 0 0 0 18 7h-1.26A8 8 0 1 0 4 15.25"
          />
        </svg>
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="5"
          />
          <line
            x1="12"
            x2="12"
            y1="1"
            y2="3"
          />
          <line
            x1="12"
            x2="12"
            y1="21"
            y2="23"
          />
          <line
            x1="4.22"
            x2="5.64"
            y1="4.22"
            y2="5.64"
          />
          <line
            x1="18.36"
            x2="19.78"
            y1="18.36"
            y2="19.78"
          />
          <line
            x1="1"
            x2="3"
            y1="12"
            y2="12"
          />
          <line
            x1="21"
            x2="23"
            y1="12"
            y2="12"
          />
          <line
            x1="4.22"
            x2="5.64"
            y1="19.78"
            y2="18.36"
          />
          <line
            x1="18.36"
            x2="19.78"
            y1="5.64"
            y2="4.22"
          />
        </svg>
      </div>
      <div
        className="d-flex gap-2"
      >
        <div>
          <p>
            latitude
          </p>
          <p
            className="details"
          />
        </div>
        <div>
          <p>
            longitude
          </p>
          <p
            className="details"
          />
        </div>
        <div>
          <p>
            temp
          </p>
          <p
            className="details"
          />
        </div>
        <div>
          <p>
            humid
          </p>
          <p
            className="details"
          />
        </div>
        <div>
          <p>
            pressure
          </p>
          <p
            className="details"
          />
        </div>
      </div>
      <div
        className="d-flex gap-2"
      >
        <div>
          <p>
            wind speed
          </p>
          <p
            className="details"
          />
        </div>
        <div>
          <p>
            sunrise
          </p>
          <p
            className="details"
          />
        </div>
        <div>
          <p>
            sunset
          </p>
          <p
            className="details"
          />
        </div>
        <div>
          <p>
            feels_like
          </p>
          <p
            className="details"
          />
        </div>
      </div>
      <div
        className="d-flex gap-3 align-items-center"
      >
        <h3>
          Description
        </h3>
      </div>
    </div>
  </div>,
]
`;
